pandoc/beamer_filter.py

```diff
@@ -210,6 +210,92 @@ def language_variant_admonition ( contents ):
                    latex_box(text) +
                    '\\hspace{1cm}}}')] )
 
+###################################
+## INCLUDE SOURCE CODE FROM FILE ##
+###################################
+
+'''
+   RST "include" directive allows the inclusion of a snippet of an
+   external file, and can format it as a block of code.
+   (https://docutils.sourceforge.io/docs/ref/rst/directives.html#including-an-external-document-fragment)
+
+   HOWEVER, Pandoc does not support it!
+
+   So these subprograms allow us to simulate it using a "container" directive.
+   The format of the directive is:
+
+      .. container:: source_include <path-to-file> [option [option ...]]
+
+   Options follow the format specified in the above link. As of now, the only options 
+   supported are
+      :start-after:<string>
+         insert code starting at the line after the first occurrence of <string>
+      :end-before:<string>
+         stop inserting code at the line before the first occurrence of <string>
+         (if "start-after" is specified, only look for <string> after starting)
+      :code:<language>
+         Language to format code insertion
+'''
+
+def source_file_contents ( filename, keywords ):
+   retval = ""
+
+   start_after = ""
+   end_before = ""
+   echo_on = False
+
+   # if we're looking for a string before starting, save the string
+   if 'start-after' in keywords.keys():
+      start_after = keywords['start-after']
+   # otherwise, we start by echoing the file
+   else:
+      echo_on = True
+
+   if 'end-before' in keywords.keys():
+      end_before = keywords['end-before']
+
+   if os.path.isfile ( filename ):
```

Comment:
Quick tip: the `pathlib` package is more powerful for most usages, to try it is to adopt it! https://docs.python.org/3/library/pathlib.html